File: test-suite/src/test/java/io/micronaut/docs/httpclientexceptionbody/BooksController.java
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.docs.httpclientexceptionbody;

import io.micronaut.context.annotation.Requires;
import io.micronaut.http.HttpResponse;
import io.micronaut.http.HttpStatus;
import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;

import java.util.HashMap;
import java.util.Map;

@Requires(property = "spec.name", value = "BindHttpClientExceptionBodySpec")
//tag::clazz[]
@Controller("/books")
public class BooksController {

    @Get("/{isbn}")
    public HttpResponse find(String isbn) {
        if (isbn.equals("1680502395")) {
            Map<String, Object> m = new HashMap<>();
            m.put("status", 401);
            m.put("error", "Unauthorized");
            m.put("message", "No message available");
            m.put("path", "/books/"+isbn);
            return HttpResponse.status(HttpStatus.UNAUTHORIZED).body(m);

        }
        return HttpResponse.ok(new Book("1491950358", "Building Microservices"));
    }
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/httpclientexceptionbody/BooksController.py
from micronaut.context.annotation import Requires
from micronaut.http import HttpResponse, HttpStatus
from micronaut.http.annotation import Controller, Get

from .Book import Book


@Requires(property="spec.name", value="BindHttpClientExceptionBodySpec")
# tag::clazz[]
@Controller("/books")
class BooksController:

    @Get("/{isbn}")
    def find(self, isbn: str) -> HttpResponse:
        if isbn == "1680502395":
            body = {
                "status": 401,
                "error": "Unauthorized",
                "message": "No message available",
                "path": "/books/" + isbn,
            }
            return HttpResponse.status(HttpStatus.UNAUTHORIZED).body(body)

        return HttpResponse.ok(Book("1491950358", "Building Microservices"))
# end::clazz[]
